src/pages/item/ItemInfo.jsx

```diff
@@ -1,39 +1,121 @@
 import React, { useState, useEffect } from "react";
-import { data } from "../../data/data";
+import Swiper from "react-id-swiper";
 import { BiMinus, BiPlus, BiShoppingBag } from "react-icons/bi";
 import { FiChevronLeft } from "react-icons/fi";
 import { useParams } from "react-router";
+import { data } from "../../data/data";
 import { Link } from "react-router-dom";
+import { RiHeartAddLine } from "react-icons/ri";
+import { useFavorites } from "../../stogre/addToLikes";
 import "./ItemInfo.css";
-import { AiOutlineHeart } from "react-icons/ai";
+import { useColorChanging } from "../../stogre/colorChangeing";
+
+// ***** component start ***** //
 
 const ItemInfo = () => {
+  const colors = useColorChanging();
+
+  const favoritesContext = useFavorites();
+  const toggleFavoriteStatusHandler = (id) => {
+    const changedId = parseInt(id);
+    const isFavorite = favoritesContext.itemIsFavorite(changedId);
+    const userFavorite = data.filter((product) => product.id === changedId);
+    if (isFavorite) {
+      favoritesContext.removeFavorite(changedId);
+    } else {
+      favoritesContext.addFavorite(userFavorite);
+    }
+    console.log(favoritesContext.favorites);
+  };
   const [product, setProduct] = useState("default droduct");
```

Comment:
product will be an object, the dafault value should be an empty object not string